### DISCUSSION TITLE
ARITHMETIC OVER/UNDERFLOW IN DSCEngine tests

### DISCUSSION THREAD

**Discussion by Kuusho** [Cyfrin/foundry-full-course-cu#2381] [Q&A]
I have been trying to debug this issue from the foundry DeFi stablecoin lesson.
According to the next lesson on Fuzz Testing, the issue is coming from this `_healthFactor` function 

function _healthFactor(address user) private view returns (uint256) {
        // get total DSC minted by user
        // get total collateral value
        (
            uint256 totalSscMinted,
            uint256 collateralValueInUsd
        ) = _getAccountInformation(user);
        if (totalSscMinted == 0) return type(uint256).max;
        uint256 collateralAdjustedForThreshold = (collateralValueInUsd *
            LIQUIDATION_THRESHOLD) / LIQUIDATION_PRECISION;
        return (collateralAdjustedForThreshold * PRECISION) / totalSscMinted;
    }

    // // check health factor (if a user has sufficient collateral)
    // // revert if they don't
    function _revertIfHealthFactorIsBroken(address user) internal view {
        uint256 userHealthFactor = _healthFactor(user);
        if (userHealthFactor < MIN_HEALTH_FACTOR) {
            revert DSCEngine__BreaksHealthFactor(userHealthFactor);
        }
    }

However, the issue persists, my question for those who've made it to this lesson is:
1. Were you able to debug this yourself?
2. How did you fix the bug?
3. How much of your code did you need to change?

PS: I've been working with claude and still no luck

**Comment by EngrPips** [comment]
If the below code is your `_healthFactor` function then you are not supposed to be running into this issue anymore.

```js
function _healthFactor(address user) private view returns (uint256) {
// get total DSC minted by user
// get total collateral value
(
uint256 totalSscMinted,
uint256 collateralValueInUsd
) = _getAccountInformation(user);
if (totalSscMinted == 0) return type(uint256).max;
uint256 collateralAdjustedForThreshold = (collateralValueInUsd *
LIQUIDATION_THRESHOLD) / LIQUIDATION_PRECISION;
return (collateralAdjustedForThreshold * PRECISION) / totalSscMinted;
}
```

**Reply by Kuusho** [reply]
Unfortunately that is exactly the case. Must have broken my code while trying to find it  or my test setup is the problem

**Reply by EngrPips** [reply]
Please run your test with the `-vvvv` flag and show the stack trace of the test process.
